infrastructure/zk/src/server.ts

```diff
@@ -3,41 +3,61 @@ import * as utils from './utils';
 import * as env from './env';
 import fs from 'fs';
 import * as db from './db/db';
+import * as deploy from './deploy';
 
 import { ethers } from 'ethers';
 
-export async function core() {
+export async function core(docker: boolean) {
     prepareForcedExitRequestAccount();
 
-    await utils.spawn(
-        'cargo run --bin zksync_server --release -- --components=eth-sender,witness-generator,forced-exit,prometheus,core,rejected-task-cleaner,fetchers,prometheus-periodic-metrics'
-    );
+    if (docker) {
+        await deploy.dockerUp('server-core');
+    } else {
+        await utils.spawn(
+            'cargo run --bin zksync_server --release -- --components=eth-sender,witness-generator,forced-exit,prometheus,core,rejected-task-cleaner,fetchers,prometheus-periodic-metrics'
+        );
+    }
 }
 
-export async function web3Node() {
-    await utils.spawn('cargo run --bin zksync_server --release -- --components=web3-api');
+export async function web3Node(docker: boolean) {
+    if (docker) {
+        await deploy.dockerUp('server-web3');
+    } else {
+        await utils.spawn('cargo run --bin zksync_server --release -- --components=web3-api');
+    }
 }
 
-export async function apiNode() {
-    await utils.spawn(
-        'cargo run --bin zksync_server --release -- --components=web3-api,rest-api,rpc-api,rpc-websocket-api'
-    );
+export async function apiNode(docker: boolean) {
+    if (docker) {
+        await deploy.dockerUp('server-api');
+    } else {
+        await utils.spawn(
+            'cargo run --bin zksync_server --release -- --components=web3-api,rest-api,rpc-api,rpc-websocket-api'
+        );
+    }
 }
 
-export async function server() {
+export async function server(docker: boolean) {
     // By the time this function is run the server is most likely not be running yet
     // However, it does not matter, since the only thing the function does is depositing
     // to the forced exit sender account, and server should be capable of recognizing
     // priority operaitons that happened before it was booted
     prepareForcedExitRequestAccount();
-
-    await utils.spawn('cargo run --bin zksync_server --release');
+    if (docker) {
+        await deploy.dockerUp('server');
+    } else {
+        await utils.spawn('cargo run --bin zksync_server --release');
+    }
 }
 
-export async function genesis() {
+export async function genesis(docker: boolean) {
```

Comment:
I'd specify false as  default value, so we don't need to change all the previous usages

Reply:
updated